lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/services/FeatureService.java

```diff
@@ -529,4 +534,149 @@ public List<Feature> readFeaturesFromFeatureListFile(List<Feature> installedFeat
         }
         return installedFeatures;
     }
+
+    /**
+     * get all platforms for all features from feature json
+     * @param libertyVersion liberty version
+     * @param libertyRuntime runtime
+     * @param requestDelay request delay
+     * @param documentURI document uri link
+     * @return set of unique platforms
+     */
+    public Set<String> getAllPlatforms(String libertyVersion, String libertyRuntime, int requestDelay, String documentURI) {
+        List<Feature> features = this.getFeatures(libertyVersion, libertyRuntime, requestDelay, documentURI);
```

Comment:
Ok, I see why you are passing the libertyVersion and libertyRuntime now, in order to get all the features for the potentially installed runtime. And if no Liberty is installed, it will get the features from the cached features.json. If the version of Liberty that is installed is older than 24.0.0.8, it will find no platforms and that would be correct.

Reply:
Okay. Please resolve this